packages/main/src/TableUtils.ts

```diff
@@ -17,9 +17,14 @@ const findRowInPath = (composedPath: Array<EventTarget>) => {
 	return composedPath.find((el: EventTarget) => el instanceof HTMLElement && el.hasAttribute("ui5-table-row")) as TableRow;
 };
 
+const isFeature = <T>(element: any, identifier: string): element is T => {
```

Comment:
Probably we can integrate the existing mechanism

```ts
// TableSelection.ts
import { registerFeature } from "@ui5/webcomponents-base/dist/FeaturesRegistry.js";

registerFeature("TableSelection", TableSelection);
```

When someone imports the TableSelection.ts module, the feature is registered, then the Table can call getFeature to check if the feature is enabeled:
```ts
import { getFeature } from "@ui5/webcomponents-base/dist/FeaturesRegistry.js";

const TableSelection = getFeature<typeof TableSelection>("TableSelection");
```

Reply:
Is it the same though?
In case of Input.ts and InputSuggestions.ts, when the import for InputSuggestions exist, the Input will create a new instance of it.

While in the Table.ts, even if you have TableSelection imported, you need to explicitly slot it into features slot of the table. This would allow you to have multiple tables with one having selection and the other not.

So even if I check whether the feature has been imported:
```
// Table.ts
const TableSelection = getFeature<typeof TableSelection>("TableSelection");
if (!TableSelection) { return; }
...
```

I still need to check, if the features slot contains a feature, that is of instance TableSelection like:
```
// Table.ts
const TableSelection = getFeature<typeof TableSelection>("TableSelection");
if (!TableSelection) { return; }

this.features.find(feature => feature instanceof TableSelection);
```

So there is no real benefit from registering the feature, isn't it?
This `isFeature` check still needs to be done to 100% determine that the feature is of type TableSelection (as to avoid this instanceof check)

So as far as I understand, this feature registration is quite nice for features that are then created by the component themselves (e.g. Input creates a new instance of InputSuggestion), but not really applicable, if the "feature" comes from the application.

Or am I misunderstanding something here?